src/material-experimental/mdc-list/list-base.ts

```diff
@@ -28,47 +34,50 @@ function toggleClass(el: Element, className: string, on: boolean) {
   }
 }
 
-@Directive()
-/** @docs-private */
-export abstract class MatListBase {
-  // @HostBinding is used in the class as it is expected to be extended. Since @Component decorator
-  // metadata is not inherited by child classes, instead the host binding data is defined in a way
-  // that can be inherited.
-  // tslint:disable-next-line:no-host-decorator-in-concrete
-  @HostBinding('class.mdc-list--non-interactive')
-  _isNonInteractive: boolean = false;
-}
-
 @Directive()
 /** @docs-private */
 export abstract class MatListItemBase implements AfterContentInit, OnDestroy, RippleTarget {
+  @HostBinding('tabindex')
+  _tabIndex = -1;
+
   lines: QueryList<ElementRef<Element>>;
 
   rippleConfig: RippleConfig = {};
 
   // TODO(mmalerba): Add @Input for disabling ripple.
   rippleDisabled: boolean;
 
+  _element: HTMLElement;
+
   private _subscriptions = new Subscription();
 
   private _rippleRenderer: RippleRenderer;
 
-  constructor(protected _element: ElementRef, protected _ngZone: NgZone, listBase: MatListBase,
-              platform: Platform) {
-    const el = this._element.nativeElement;
+  protected constructor(private _elementRef: ElementRef<HTMLElement>, protected _ngZone: NgZone,
+                        listBase: MatListBase, platform: Platform) {
+    this._element = this._elementRef.nativeElement;
     this.rippleDisabled = listBase._isNonInteractive;
     if (!listBase._isNonInteractive) {
-      el.classList.add('mat-mdc-list-item-interactive');
+      this._element.classList.add('mat-mdc-list-item-interactive');
     }
     this._rippleRenderer =
-        new RippleRenderer(this, this._ngZone, el, platform);
-    this._rippleRenderer.setupTriggerEvents(el);
+        new RippleRenderer(this, this._ngZone, this._element, platform);
+    this._rippleRenderer.setupTriggerEvents(this._element);
   }
 
   ngAfterContentInit() {
     this._monitorLines();
   }
 
+  ngOnDestroy() {
+    this._subscriptions.unsubscribe();
+    this._rippleRenderer._removeTriggerEvents();
+  }
+
+  _setTabIndexForChildren(value: number) {
+    this._element.querySelectorAll('a, button').forEach(el => (el as HTMLElement).tabIndex = value);
```

Comment:
Isn't this going to clobber any tabindex that the consumer might've set?

Reply:
yeah, I commented about that too (see above)